symengine/dict.h

```diff
@@ -42,6 +43,7 @@ typedef std::map<RCP<const Integer>, unsigned,
         RCPIntegerKeyLess> map_integer_uint;
 typedef std::map<unsigned, integer_class>
        map_uint_mpz;
+typedef std::map<int, Expression> map_int_Expr;
```

Comment:
@isuruf , why have we used `Expression` instead of `Basic`? Isn't `Basic` the main building block and the most general class? `Expression` is used just as an upper layer wrapper for `Basic`, right? Shouldn't we have `Basic` here, to be faster (and not depend on function overloads, like in `sub_uni_poly`)?